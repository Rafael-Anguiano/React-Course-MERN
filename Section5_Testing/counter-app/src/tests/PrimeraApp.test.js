import React from 'react';
import '@testing-library/jest-dom'
// import { render } from "@testing-library/react"
import {shallow} from 'enzyme'
import PrimeraApp from '../FirstApp'

describe('Tests in PrimeraApp file', () => {
    // test('should show the message <Hello Im Goku>', () => {
    //     const greet = "Hello Im Goku";
    //     const { getByText } = render(<PrimeraApp greet={greet} />);
    
    //     expect(getByText(greet)).toBeInTheDocument();
    // })
    
    test('should show PrimeraApp correctly', () => {
        const greet = "Hello Im Goku";
        const wrapper = shallow(<PrimeraApp greet={greet} />)
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show Subtitle sent by props', () => {
        const greet = "Hello Im Goku";
        const subtitle = "Im a subtitle";
        const wrapper = shallow(
            <PrimeraApp 
                greet={greet} 
                subtitle={subtitle}
            />
        )
        const textParagraph = wrapper.find('p').text()
        expect(textParagraph).toBe(subtitle)

    })
})
