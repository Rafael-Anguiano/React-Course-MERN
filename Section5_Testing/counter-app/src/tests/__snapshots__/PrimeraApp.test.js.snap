// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in PrimeraApp file should show PrimeraApp correctly 1`] = `
<Fragment>
  <h1>
    Hello Im Goku
  </h1>
  <p>
    Im a default subtitle
  </p>
</Fragment>
`;
